packages/react-query-next-experimental/src/HydrationStreamProvider.tsx

```diff
@@ -143,31 +143,28 @@ export function createHydrationStreamProvider<TShape>() {
       onEntriesRef.current = props.onEntries
     })
 
-    React.useEffect(() => {
+    // Setup and run the onEntries handler on the client only, but do it during
+    // the initial render so children have access to the data immediately
+    // This is important to avoid the client suspending during the initial render
+    // if the data has not yet been hydrated.
+    // @ts-expect-error window isn't defined on the server
+    if (typeof window !== 'undefined' && !window[id]?.initialized) {
```

Comment:
we have an `isServer` util function that works slightly different than just checking for window. could we re-use that one ?

Reply:
Done!